### PR TITLE
Add script to export town close QC reports and print command for refreshing tables

### PR DESCRIPTION
**Pull request by jeancochrane**
This PR adds a new script `dbt/scripts/export_qc_town_close_reports.py` with two primary functions:

1. Exports the QC town close reports
2. When the `--print-table-refresh-command` argument is provided, prints a command that can be run using the [`service-spark-iasworld`](https://github.com/ccao-data/service-spark-iasworld/) project to refresh data for iasWorld tables that the report depends on, and quits instead of exporting reports

Closes https://github.com/ccao-data/data-architecture/issues/576.

### CODE REVIEW

File: dbt/scripts/export_qc_town_close_reports.py [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+    "39": "3",
+    "70": "1",
+    "71": "1",
+    "72": "1",
+    "73": "1",
+    "74": "1",
+    "75": "1",
+    "76": "1",
+    "77": "1",
+}
```
Comment (jeancochrane) [inline dbt/scripts/export_qc_town_close_reports.py:84]: Once again, I long for a Python version of the `ccao` package...

File: dbt/scripts/export_qc_town_close_reports.py [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+                formatted_dep["min_year"] = formatted_dep["max_year"] = None
+            if table_name == "asmt_hist":
+                # The asmt_hist table doesn't have any cur = 'Y' records
+                del formatted_dep["cur"]
+
+            iasworld_deps[table_name] = formatted_dep
+
+        print("ssh into the server and run the following commands:")
+        print()
+        print("cd /home/shiny-server/services/service-spark-iasworld")
```
Comment (jeancochrane) [inline dbt/scripts/export_qc_town_close_reports.py:210]: Are you fine with storing this path under version control?

File: socrata/socrata_upload.py [hunk 3]
```diff
@@ -1,11 +1,12 @@
...
 import io
 import json
+import os
+import time
+
 import pandas as pd
+import requests
+from dbt.cli.main import dbtRunner
 from pyathena import connect
 from pyathena.pandas.cursor import PandasCursor
```
Comment (jeancochrane) [inline socrata/socrata_upload.py:12]: This diff is unrelated to the PR, and is simply a workaround to resolve the linting issue that is also fixed by https://github.com/ccao-data/data-architecture/pull/587.

File: dbt/README.md [hunk 4]
```diff
@@ -686,65 +686,128 @@ section with documentation once that effort is resolved.
...
+* **`--target`** (optional): The name of the [dbt
+  target](https://docs.getdbt.com/reference/dbt-jinja-functions/target) to run
+  queries against. Defaults to `dev`. You should never have to set this option
+  when running the script locally.
+
+Other scripts that build on `export_models` for specific workflows tend to
+expose similar options. See the documentation for these workflows below for
+more details.
+
+#### Running town close QC reports with `export_qc_town_close_reports`
```
[the remark this answers is not in the document]
Comment (dfsnow) [reply dbt/README.md:732]: I prefer the current name tbh.

File: dbt/README.md [hunk 5]
```diff
@@ -686,65 +686,128 @@ section with documentation once that effort is resolved.
...
+
+We run town close reports using the [`scripts/export_qc_town_close_reports.py`
+script](./scripts/export_qc_town_close_reports.py), which builds on top of
+`export_models`. As such, `export_qc_town_close_reports` expects the same set
+of Python requirements as `export_models`, which can be installed in a virtual
+environment by running `pip install scripts/requirements.export_mod.els.txt`.
```
Comment (dfsnow) [inline dbt/README.md:738]:
```suggestion
environment by running `pip install scripts/requirements.export_models.txt`.
```

File: dbt/README.md [hunk 6]
```diff
@@ -686,65 +686,128 @@ section with documentation once that effort is resolved.
...
+* **`--year`** (optional): The year to use for filtering results. Defaults to the current year.
+* **`--target`** (optional): The name of the [dbt
+  target](https://docs.getdbt.com/reference/dbt-jinja-functions/target) to run
+  queries against. Defaults to `dev`. You should never have to set this option
+  when running the script locally.
+* **`--rebuild` or `--no-rebuild`** (optional): This flag determines whether or not the script will rebuild
+  the selected models using `dbt run` prior to export. It defaults to false (`--no-rebuild`) and
+  is most useful in rare cases where the underlying models that comprise the reports have been
+  edited since the last run, typically during the period when a QC report is under active development.
+* **`--print-table-refresh-command`** (optional): Instructs the script to print a command that can be
```
[the remark this answers is not in the document]
Comment (dfsnow) [reply dbt/README.md:755]: I think it's fine!

File: dbt/scripts/export_models.py [hunk 7]
```diff
@@ -87,20 +88,23 @@ def main():
...
-    select = args.select
-    selector = args.selector
-    rebuild = args.rebuild
-    where = args.where
+    return parser.parse_args()
 
+
+def export_models(
+    target: str = "dev",
+    select: typing.List[str] | None = None,
```
Comment (dfsnow) [inline dbt/scripts/export_models.py:96]: **question (non-blocking):** Why not use the new built-ins?

```suggestion
    select: list[str] | None = None,
```

File: dbt/scripts/export_qc_town_close_reports.py [hunk 8]
```diff
@@ -0,0 +1,228 @@
...
+import io
+import json
+import os
+import sys
+
+from dbt.cli.main import dbtRunner
+
+# Add the parent directory of `scripts` to the module search path
+# so that we can import from other modules in the `scripts` directory
+sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
```
[the remark this answers is not in the document]
Comment (dfsnow) [reply dbt/scripts/export_qc_town_close_reports.py:16]: Hmmmm, yeah this is pretty wonky. Why not move some of the export functionality into a `utils/` dir/module. Then `dbt/scripts/export_models.py` can be a simple wrapper for the `utils/` code, and both `export_models.py` and `export_qc_town_close_reports.py` can share the utils code.

File: dbt/scripts/export_qc_town_close_reports.py (hunk 2 above)
Comment (dfsnow) [reply dbt/scripts/export_qc_town_close_reports.py:210]: Yep!

File: dbt/scripts/export_qc_town_close_reports.py [hunk 9]
```diff
@@ -0,0 +1,228 @@
...
+            "--resource-types",
+            "source",
+            "--output",
+            "json",
+            "--output-keys",
+            "name",
+            "source_name",
+            "--select",
+            f"+{tag}",
+            f"+{tag}_{tag_suffix}",
```
Comment (dfsnow) [inline dbt/scripts/export_qc_town_close_reports.py:168]: **question (non-blocking):** Do we need to worry about the `indirect_selection: cautious` that the selector used?

**Review by dfsnow — APPROVED** [review]
Excited that this is ready to go! However, a little bit wary because the different export scripts seem to be multiplying and the copy-then-paste-a-command abstraction feels pretty hacky. IIRC, the plan is to get something useable before improving on it later, so let's sit down and brainstorm next steps before I'm outta here in late Sept.

File: dbt/README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (jeancochrane) [reply dbt/README.md:732]: Great, I'll keep it as-is then.

File: dbt/README.md (hunk 5 above)
Comment (jeancochrane) [reply dbt/README.md]: Oops, thanks!

File: dbt/README.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (jeancochrane) [reply dbt/README.md:755]: Great, keeping it the same for now, then.

File: dbt/scripts/export_qc_town_close_reports.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (jeancochrane) [reply dbt/scripts/export_qc_town_close_reports.py:16]: That makes sense! If it's cool with you, I'll do that in a fast-follow PR so that you can review it with confidence that no other parts of the code have changed during the file moving process.

File: dbt/scripts/export_qc_town_close_reports.py (hunk 9 above)
Comment (jeancochrane) [reply dbt/scripts/export_qc_town_close_reports.py:168]: Great question! Per [the docs on indirect selection](https://docs.getdbt.com/reference/global-configs/indirect-selection), we shouldn't need it because it only applies to the `build` and `run` commands:

> Use the `--indirect_selection` flag to `dbt test` or `dbt build` to configure which tests to run for the nodes you specify.

File: dbt/scripts/export_models.py (hunk 7 above)
Comment (jeancochrane) [reply dbt/scripts/export_models.py]: Embarrassingly, it was just because I didn't know about the new bulitins as of 3.9 😅 Fixed in 785ee5437f1b8ae616a30db1971dcd5535997de2 so that I could also remove the now-unused `typing` import.
